.github/changeset-version.js

```diff
@@ -77,46 +100,56 @@ function main() {
-		// Replace first incorrect version, it should be the version header
-		changelog = changelog.replace(miniflareVersion, nextMiniflareVersion);
-		fs.writeFileSync(miniflareChangelogPath, changelog);
-
 		// ...update `miniflare`'s `package.json` version
 		miniflarePkg.version = nextMiniflareVersion;
 		setPkg(miniflarePkgPath, miniflarePkg);
 
-		// ...update `miniflare` version in dependencies of other packages
-		// (this next command returns a list of absolute package paths in the
-		// workspace separated by newline)
-		const listResult = execSync("pnpm list --recursive --depth -1 --parseable");
-		const paths = listResult.toString().trim().split("\n");
-		for (const p of paths) {
-			// Ignore warning lines
-			if (!path.isAbsolute(p)) continue;
-			const pkgPath = path.join(p, "package.json");
-			const pkg = getPkg(pkgPath);
-			let changed = false;
-			for (const key of [
-				"dependencies",
-				"devDependencies",
-				"peerDependencies",
-				"optionalDependencies",
-			]) {
-				const constraint = pkg[key]?.["miniflare"];
-				if (constraint === undefined) continue;
-				// Don't update `workspace:`-style constraints
-				if (constraint.startsWith("workspace:")) continue;
-				pkg[key]["miniflare"] = nextMiniflareVersion;
-				changed = true;
+		const changedPathsBuffer = execSync("git ls-files --modified", {
+			cwd: rootPath,
+		});
+		const changedPaths = changedPathsBuffer.toString().trim().split("\n");
+		for (const relativeChangedPath of changedPaths) {
+			const changedPath = path.resolve(rootPath, relativeChangedPath);
+			const name = path.basename(changedPath);
+			if (name === "package.json") {
+				// ...update `miniflare` version in dependencies of other packages
+				const pkg = getPkg(changedPath);
+				let changed = false;
+				for (const key of [
+					"dependencies",
+					"devDependencies",
+					"peerDependencies",
+					"optionalDependencies",
+				]) {
+					const constraint = pkg[key]?.["miniflare"];
+					if (constraint === undefined) continue;
+					// Don't update `workspace:`-style constraints
+					if (constraint.startsWith("workspace:")) continue;
+					pkg[key]["miniflare"] = nextMiniflareVersion;
+					changed = true;
+				}
+				if (changed) setPkg(changedPath, pkg);
+			} else if (name === "CHANGELOG.md") {
+				// ...update `CHANGELOG.md`s with correct version
+				await transformFile(changedPath, (line) => {
+					// Replace version header in `miniflare` `CHANGELOG.md`
+					line = line.replace(
+						`## ${miniflareVersion}`,
+						`## ${nextMiniflareVersion}`
+					);
+					// Replace `Updated dependencies` line in other `CHANGELOG.md`s
+					line = line.replace(
+						`- miniflare@${miniflareVersion}`,
+						`- miniflare@${nextMiniflareVersion}`
+					);
+					return line;
+				});
 			}
-			if (changed) setPkg(pkgPath, pkg);
 		}
 	}
 
 	// 4. Update the lockfile
 	execSync("pnpm install");
 }
 
-if (require.main === module) main();
+if (require.main === module) void main();
```

Comment:
Disregarding my other comment, If we keep this async, we should at least catch errors and log them out